components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/resources/web/enrich-mediator/update-mediator.jsp

```diff
@@ -67,11 +84,22 @@
 
     if (request.getParameter("mediator.enrich.target.val_ex") != null && !request.getParameter("mediator.enrich.target.val_ex").trim().equals("")) {
         if (request.getParameter("mediator.enrich.target.type").equals("custom")) {
-            XPathFactory xPathFactory2 = XPathFactory.getInstance();
-            enrichMediator.setTargetExpression(xPathFactory2.createSynapseXPath("mediator.enrich.target.val_ex",
-                    request.getParameter("mediator.enrich.target.val_ex"),session));
+            String expression = request.getParameter("mediator.enrich.target.val_ex");
+            if (expression.startsWith(JSON_PATH)) {
+                int expLength = expression.length();
+                String extractedExp = expression.substring(JSON_PATH.length(), expLength - 1);
+                SynapseJsonPathFactory synapseJsonPathFactory = SynapseJsonPathFactory.getInstance();
+                enrichMediator.setTargetExpression(synapseJsonPathFactory.createSynapseJsonPath(
+                        "mediator.enrich.target.val_ex", extractedExp));
+            } else {
+                XPathFactory xPathFactory2 = XPathFactory.getInstance();
+                enrichMediator.setTargetExpression(xPathFactory2.createSynapseXPath("mediator.enrich.target.val_ex",
+                        expression, session));
+            }
+            enrichMediator.setTargetProperty(null);
```

Comment:
Can we avoid the duplicate code here?

Reply:
It's not recommended to write methods inside JSP pages.